service/history/resource/resource.go

```diff
@@ -72,6 +72,7 @@ func New(
 			serviceResource.GetHistoryManager(),
 			params.Logger,
 			params.MetricsClient,
+			uint(config.EventsCacheMaxSizeInKb()),
```

Comment:
would be easier just uint64 everywhere for cache size.
For config, it also might be better to also use bytes to be consistent with rest